src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -576,9 +577,31 @@ protected RubyDynamicObject clone(RubyDynamicObject object, boolean freeze,
                 newObjectMetaClass.fields.initCopy(selfMetaClass);
             }
 
-            initializeCloneNode.call(newObject, "initialize_clone", object);
+            // pass :freeze keyword argument to #initialize_clone
+            if (toForceFreezing(freeze) || toForceUnfreezing(freeze)) {
+                final String string = "freeze";
+                final LeafRope rope = RopeOperations.encodeAscii(string, USASCIIEncoding.INSTANCE);
+                final RubySymbol key = new RubySymbol(string, rope, Encodings.US_ASCII);
```

Comment:
The RubySymbol should be package-private so it's clear it shouldn't be used directly.
There is already `coreSymbols().FREEZE` which you can use :)

Reply:
Thank you. It fixed the failed specs as well 🎆